From: Francois Romieu <romieu@fr.zoreil.com>
Date: Fri, 11 Mar 2011 20:30:24 +0100
Subject: [PATCH 62/71] r8169: rtl8169_set_speed_xmii cleanup.

commit 826e6cbdadfa51495c7189641df2514cc48e23da upstream.

Shorten chipset version test.

No functional change.

Careful readers will notice that the 'supports_gmii' flag is deduced
from the device PCI id. Though less specific than the chipset related
RTL_GIGA_MAC_VER_XY, it is good enough to detect a GMII deprieved 810x.
Some features push for a device specific configuration (improved jumbo
frame support for instance). 'supports_gmii' will follow this path
if / when the device PCI id test stops working.

Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
Cc: Realtek linux nic maintainers <nic_swsd@realtek.com>
---
 drivers/net/r8169.c |   11 +----------
 1 files changed, 1 insertions(+), 10 deletions(-)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index ed8cf86..4ad6ba3 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -1261,16 +1261,7 @@ static int rtl8169_set_speed_xmii(struct net_device *dev,
 		giga_ctrl &= ~(ADVERTISE_1000FULL | ADVERTISE_1000HALF);
 
 		/* The 8100e/8101e/8102e do Fast Ethernet only. */
-		if (tp->mac_version != RTL_GIGA_MAC_VER_07 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_08 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_09 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_10 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_13 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_14 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_15 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_16 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_29 &&
-		    tp->mac_version != RTL_GIGA_MAC_VER_30) {
+		if (tp->mii.supports_gmii) {
 			if (adv & ADVERTISED_1000baseT_Half)
 				giga_ctrl |= ADVERTISE_1000HALF;
 			if (adv & ADVERTISED_1000baseT_Full)
-- 
1.7.5.4

